RotaryEncoder.h

```diff
@@ -1,50 +1,52 @@
-// -----
-// RotaryEncoder.h - Library for using rotary encoders.
-// This class is implemented for use with the Arduino environment.
-// Copyright (c) by Matthias Hertel, http://www.mathertel.de
-// This work is licensed under a BSD style license. See http://www.mathertel.de/License.aspx
-// More information on: http://www.mathertel.de/Arduino
-// -----
-// 18.01.2014 created by Matthias Hertel
-// 16.06.2019 pin initialization using INPUT_PULLUP
-// -----
-
-#ifndef RotaryEncoder_h
-#define RotaryEncoder_h
-
-#include "Arduino.h"
-
-#define LATCHSTATE 3
-
-class RotaryEncoder
-{
-public:
-  // ----- Constructor -----
-  RotaryEncoder(int pin1, int pin2);
-  
-  // retrieve the current position
-  long  getPosition();
-  
-  // simple retrieve of the direction the knob was rotated at. 0 = No rotation, 1 = Clockwise, -1 = Counter Clockwise
-  int8_t getDirection();
-
-  // adjust the current position
-  void setPosition(long newPosition);
-
-  // call this function every some milliseconds or by using an interrupt for handling state changes of the rotary encoder.
-  void tick(void);
-
-private:
-  int _pin1, _pin2; // Arduino pins used for the encoder. 
-  
-  volatile int8_t _oldState;
-  
-  volatile long _position;         // Internal position (4 times _positionExt)
-  volatile long _positionExt;      // External position
-  volatile long _positionExtPrev;  // External position (used only for direction checking)
-
-};
-
-#endif
-
+// -----
+// RotaryEncoder.h - Library for using rotary encoders.
+// This class is implemented for use with the Arduino environment.
+// Copyright (c) by Matthias Hertel, http://www.mathertel.de
+// This work is licensed under a BSD style license. See http://www.mathertel.de/License.aspx
+// More information on: http://www.mathertel.de/Arduino
+// -----
+// 18.01.2014 created by Matthias Hertel
+// 16.06.2019 pin initialization using INPUT_PULLUP
+// -----
+
+#ifndef RotaryEncoder_h
+#define RotaryEncoder_h
+
+#include "Arduino.h"
+
+#define LATCHSTATE 3
+
+class RotaryEncoder
+{
+public:
+  enum class Direction { NOROTATION = 0, CLOCKWISE = 1, COUNTERCLOCKWISE = 2};
```

Comment:
Direction::COUNTERCLOCKWISE should be -1 to be compatible with the current version.

Reply:
You are right. An additional commit to fix this is available.